### PR TITLE
Fix pvc requests.storage validation

### PR DESCRIPTION
**Pull request by derekwaynecarr**
A `PersistentVolumeClaim` should not be able to request a negative amount of storage.

/cc @kubernetes/sig-storage @kubernetes/rh-cluster-infra @deads2k

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30386)

### CODE REVIEW

**Comment by deads2k** [comment]
lgtm

**Comment by derekwaynecarr** [comment]
@k8s-bot test this issue https://github.com/kubernetes/kubernetes/issues/30377

**Comment by jsafrane** [comment]
```
!!! 'gofmt -s -w' needs to be run on the following files: 
./pkg/api/validation/validation_test.go
FAILED   hack/make-rules/../../hack/verify-gofmt.sh	11s
```

**Comment by derekwaynecarr** [comment]
Let's see if gofmt gods are happy now.  I have no idea why vscode was not well-behaved on the first pass.  Maybe that file is just too big ;-)

**Comment by derekwaynecarr** [comment]
@k8s-bot test this issue https://github.com/kubernetes/kubernetes/issues/27235

**Comment by ncdc** [comment]
@smarterclayton ptal

**Comment by smarterclayton** [comment]
Changes looked fine to me

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -1014,26 +1014,37 @@ func ValidatePersistentVolumeStatusUpdate(newPv, oldPv *api.PersistentVolume) fi
...
-		allErrs = append(allErrs, field.Required(specPath.Child("accessModes"), "at least 1 accessMode is required"))
+	allErrs = append(allErrs, ValidatePersistentVolumeClaimSpec(&pvc.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidatePersistentVolumeClaimSpec validates a PersistentVolumeClaimSpec
+func ValidatePersistentVolumeClaimSpec(spec *api.PersistentVolumeClaimSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(spec.AccessModes) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("accessModes"), "at least 1 accessMode is required"))
```
Comment (thockin) [inline pkg/api/validation/validation.go]: nit: s/accessMode/ access mode/ - the field name is already referenced

**Comment by derekwaynecarr** [comment]
Addressed nit.

**Comment by deads2k** [comment]
lgtm

**Comment by derekwaynecarr** [comment]
@k8s-bot test this https://github.com/kubernetes/kubernetes/issues/27462

**Comment by derekwaynecarr** [comment]
@k8s-bot test this issue https://github.com/kubernetes/kubernetes/issues/30570

**Comment by jsafrane** [comment]
@k8s-bot test this issue #30570

**Pull request merged**